packages/lint-eslint-config-rules/src/cli.ts

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+import { readFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { parseArgs } from 'node:util';
+
+import { lintESLintConfigRules, sortedRuleIds } from './index.js';
+
+export interface IESLintConfigRulesLintResultJSON {
+  knownRuleIds: string[];
+  usedRuleIds: string[];
+  usedPluginSpecifiers: string[];
+  usedDeprecatedRuleIds: string[];
+  usedUnknownRuleIds: string[];
+}
+
+const main = async () => {
+  const args = parseArgs({
+    options: {
+      cwd: {
+        type: 'string',
+        default: process.cwd(),
+      },
+      help: {
+        type: 'boolean',
+        short: 'h',
+        default: false,
+      },
+      json: {
+        type: 'boolean',
+        default: false,
+      },
+    },
+  });
+
+  const { cwd, help, json } = args.values as {
+    [key in keyof typeof args.values]: NonNullable<(typeof args.values)[key]>;
+  };
+
+  if (help) {
+    // MEMO: use JSON modules when it's stable to simplify this
+    // https://nodejs.org/api/esm.html#json-modules
+    const packageJson = JSON.parse(
+      await readFile(join(import.meta.dirname, '..', 'package.json'), {
+        encoding: 'utf8',
+      }),
+    ) as { name: string; version: string; description: string };
+    console.log(`${packageJson.name} v${packageJson.version}
+${packageJson.description}
+
+Usage: lint-eslint-config-rules [options]
+
+Options:
+  --cwd <path>\tThe directory to lint (default: process.cwd())
+  --json\tOutput all information in JSON format
+  -h, --help\tDisplay this help message
+
+Note:
+  "used" means the rule is specified in the config(including all extended configs), whether it's "off" or "warn" or "error".
+`);
+    return;
+  }
+
+  if (!json) {
+    console.log(`Checking ESLint rules in ${cwd}`);
+  }
```

Comment:
nit:

```suggestion
  console.error(`Checking ESLint rules in ${cwd}`);
```

Reply:
`console.error` will print red text in terminal, we may consider adding chalk to better format the output in the future 🤔

Reply:
Maybe we need to abstract a `logger`, and use package like [picocolors](https://www.npmjs.com/package/picocolors) again.

```
import { red  } from 'picocolors';

console.error(red('ERROR') + ' ', ...),
```